File: Data Structures & Algorithms/top-k-elements-in-list/submission-4.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        counts={}
        for num in nums:
            counts[num] = counts.get(num, 0)+1

        sorCounts = sorted(counts.items(), key=lambda x: x[1], reverse=True)
        sol = []

        print(sorCounts)
        for i in range(k):
            sol.append(sorCounts[i][0])
        
        return sol

        
        

File: Data Structures & Algorithms/duplicate-integer/submission-4.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {

        std::map<int, int> counts;

        for(auto i : nums){
            counts[i]++;
            if(counts[i] > 1)
                return true;
        }

        return false;
    }
};